common/app/uiComponent/core/JavascriptEngine.scala

```diff
@@ -0,0 +1,25 @@
+package uiComponent.core
+
+import java.io.Reader
+import javax.script._
+
+import jdk.nashorn.api.scripting.JSObject
+
+import scala.util.Try
+
+class JavascriptEngine() {
+  import JavascriptEngine._
+
+  private lazy val bindings = shared.createBindings()
```

Comment:
I had read it. That's why I made `JavascriptEngine` a class with its own bindings, to avoid mixing scripts evaluation.

I made it a bit simpler now by making `JavascriptEngine` a pure object where functions take a `SimpleScriptContext` as a implicit param. https://github.com/guardian/frontend/pull/17141/files#diff-6a1a33938043036ec3d005e3fc9107afR11
Each `JavascriptRenderer` will create its own `SimpleScriptContext`